frontend/apps/reader/modules/readertypography.lua

```diff
@@ -0,0 +1,802 @@
+                -- feat = feat .. "       "
+                text = text .. "   " .. feat
+                if lang_tag == G_reader_settings:readSetting("text_lang_default") then
+                    text = text .. "   ★"
+                end
+                if lang_tag == G_reader_settings:readSetting("text_lang_fallback") then
+                    text = text .. "   �"
+                end
+                return text
+            end,
+            callback = function()
+                UIManager:show(InfoMessage:new{
+                    text = T(_("Changed typography language to %1."), BD.wrap(lang_name)),
+                })
+                self.text_lang_tag = lang_tag
+                self.ui.document:setTextMainLang(lang_tag)
+                self.ui:handleEvent(Event:new("UpdatePos"))
+            end,
+            hold_callback = function(touchmenu_instance)
+                UIManager:show(MultiConfirmBox:new{
+                    -- No real need for a way to remove default one, we can just
+                    -- toggle between setting a default OR a fallback (if a default
+                    -- one is set, no fallback will ever be used - if a fallback one
+                    -- is set, no default is wanted; so when we set one below, we
+                    -- remove the other).
+                    text = T( _("Would you like %1 to be used as the default (★) or fallback (�) typography language?\n\nDefault will always take precedence while fallback will only be used if the language of the book can't be automatically determined."), BD.wrap(lang_name)),
+                    choice1_text = _("Default"),
+                    choice1_callback = function()
+                        G_reader_settings:saveSetting("text_lang_default", lang_tag)
+                        G_reader_settings:delSetting("text_lang_fallback")
+                        if touchmenu_instance then touchmenu_instance:updateItems() end
+                    end,
+                    choice2_text = C_("Hyphenation", "Fallback"),
+                    choice2_callback = function()
+                        G_reader_settings:saveSetting("text_lang_fallback", lang_tag)
+                        G_reader_settings:delSetting("text_lang_default")
+                        if touchmenu_instance then touchmenu_instance:updateItems() end
+                    end,
+                })
+            end,
+            checked_func = function()
+                return self.text_lang_tag == lang_tag
+            end,
+        })
+    end
+
+    self.language_submenu.max_per_page = 5
+    table.insert(self.menu_table, {
+        text_func = function()
+            local lang_name = LANG_TAG_TO_LANG_NAME[self.text_lang_tag] or self.text_lang_tag
+            return T(_("Typography language: %1"), lang_name)
+        end,
+        sub_item_table = self.language_submenu,
+    })
+
+    table.insert(self.menu_table, {
+        text = _("Follow document embedded lang tags"),
```

Comment:
document's

Or simply `Honor embedded lang tags`?

Reply:
Something like `use` instead of `honor` is probably a bit easier to understand.

Reply:
Fair enough ;).

Reply:
`Use embedded language tags` ?
`Use document's embedded language tags` ?
`language tags` (for gramatically english) or `lang tags` (for HTMLically english :) ?

I'm not fond of the word "honor", but it's clearer that just "use". "Use" is a bit vague.
It's not like we "use" them, we "use" something else that will do according to them.
We don't _use_ road signs :)

Reply:
You could just leave out the verb; the checkmark already says `use`/`honor`/etc.

Reply:
`[ ] Document's embedded lang tags` looks very very strange and incomplete to me :)